src/video/allegro_v.cpp

```diff
@@ -40,14 +40,13 @@ static FVideoDriver_Allegro iFVideoDriver_Allegro;
 
 static BITMAP *_allegro_screen;
 
-#define MAX_DIRTY_RECTS 100
-static PointDimension _dirty_rects[MAX_DIRTY_RECTS];
-static int _num_dirty_rects;
+static PointDimension _dirty_rects[100];
+static size_t _num_dirty_rects;
 static Palette _local_palette; ///< Current palette to use for drawing.
 
 void VideoDriver_Allegro::MakeDirty(int left, int top, int width, int height)
 {
-	if (_num_dirty_rects < MAX_DIRTY_RECTS) {
+	if (_num_dirty_rects < lengthof(_dirty_rects)) {
```

Comment:
`std::size(_dirty_rects)` though :)

Reply:
I tried replacing `lengthof` with `std::size`. But it does not work... 

Despite `std::size` being `constexpr` it cannot be used as compile time constant.
* Usages in `static_assert` no longer compile.
* Things like 
  ```c++
   CargoType accepts[lengthof(hs->accepts_cargo)];
   ```
  become dynamically sized stack arrays.

Some places use `std::tuple_size_v` instead, but that is more complicated.

Reply:
It should work *here* though, surely? I'm not saying replace all existing lengthof, just don't add them.